src/main/java/io/strimzi/kafka/quotas/StaticQuotaCallback.java

```diff
@@ -103,47 +104,54 @@ public void configure(Map<String, ?> configs) {
         storageCheckInterval = config.getStorageCheckInterval();
         logDirs = config.getLogDirs();
 
-        storageChecker.start();
         log.info("Configured quota callback with {}. Storage quota (soft, hard): ({}, {}). Storage check interval: {}", quotaMap, storageQuotaSoft, storageQuotaHard, storageCheckInterval);
     }
 
     private class StorageChecker implements Runnable {
         private final Thread storageCheckerThread = new Thread(this, "storage-quota-checker");
-        private volatile boolean running = false;
+        private AtomicBoolean running = new AtomicBoolean(false);
 
-        void start() {
-            if (!running) {
-                running = true;
+        void startIfNecessary() {
+            if (!running.get()) {
+                running.set(true);
```

Comment:
The point of changing to `AtomicBoolean` is so we can use an atomic compare and swap (CAS) operation, so rather than using `get` and `set`, which together won't be atomic, you need to use `compareAndSet(boolean expectedValue, boolean newValue)`, which will be.